webauth/agent.js

```diff
@@ -17,10 +17,12 @@ export default class Agent {
         resolve(event.url);
       };
       Linking.addEventListener('url', urlHandler);
-      NativeModules.A0Auth0.showUrl(url, closeOnLoad, err => {
+      NativeModules.A0Auth0.showUrl(url, closeOnLoad, result => {
         Linking.removeEventListener('url', urlHandler);
-        if (err) {
-          reject(err);
+        if (result && result.url) {
```

Comment:
This seems odd ... if there is a `result`, it could be a success or an error? Is there maybe an `error_code` property on there when it's an error? Just seems like a non-standard way to pass a result/response to a callback.

Reply:
So I thought about this at the time, I could change the result to have a status e.g. success or error. Then you're changing the response to the end user as some errors have just been passed straight out from the Native Platform iOS/Android into err. Although looking at certainly iOS errors I don't think the JS side is equipped to process them anyway.

Perhaps we should more strongly define the errors from the native platforms or we need to change the callback method.
Currently using `RCTResponseSenderBlock` which only accept 1 argument which is an array.
https://facebook.github.io/react-native/docs/native-modules-ios

It might be better to use `RCTPromiseResolveBlock` although this becomes more of a refactor in both JS/iOS/Android to achieve this.